From: Brad Peters <bpeters@redhat.com>
Date: Wed, 5 Mar 2008 17:18:55 -0500
Subject: [scsi] ibmvscsi: set command timeout to 60 seconds
Message-id: 20080305221855.GC8084@bpeters-ibm
O-Subject: Re: [RHEL 5.2 PATCH 1/1] : #354611: Can't install RHEL5.1-Beta using a multi-path device.
Bugzilla: 354611

Set the default command timeout for ibmvscsi disks to 60 seconds
to ensure we don't prematurely timeout commands. This fixes a problem
seen where the default 30 seconds was not long enough due to
congestion on the server.

Signed-off-by: Brian King <brking@linux.vnet.ibm.com>

Acked-by: Doug Ledford <dledford@redhat.com>

diff --git a/drivers/scsi/ibmvscsi/ibmvscsi.c b/drivers/scsi/ibmvscsi/ibmvscsi.c
index 537e806..ac118a7 100644
--- a/drivers/scsi/ibmvscsi/ibmvscsi.c
+++ b/drivers/scsi/ibmvscsi/ibmvscsi.c
@@ -1395,8 +1395,10 @@ static int ibmvscsi_slave_configure(struct scsi_device *sdev)
 	unsigned long lock_flags = 0;
 
 	spin_lock_irqsave(shost->host_lock, lock_flags);
-	if(sdev->type == TYPE_DISK)
+	if(sdev->type == TYPE_DISK) {
 		sdev->allow_restart = 1;
+		sdev->timeout = 60 * HZ;
+	}
 	scsi_adjust_queue_depth(sdev, 0, shost->cmd_per_lun);
 	spin_unlock_irqrestore(shost->host_lock, lock_flags);
 	return 0;
